Modules/Product/Http/Controllers/ProductController.php

```diff
@@ -2,665 +2,292 @@
-        Video::where('product_id', $req->id)->delete();
-        Redis::flushDB();
-        $response = ['res' => true, 'msg' => "Deleted Successfully", 'data' => ""];
+        $user = auth()->user();
+
+        $image = Product::where('id', $request->product_id)->where('user_id', $user->id)->first();
+        if ($user->cannot('delete', $image)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+        $response = $this->productService->deleteImage($request);
 
         return response()->json($response);
     }
 
-    public function deleteProductImage(Request $req)
+    /**
+     * Delete product video by respected product and image id
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function deleteVideo(Request $request): JsonResponse
     {
-        ProductImage::where('id', $req->image_id)->delete();
-        $response = ['res' => true, 'msg' => "Deleted Successfully", 'data' => ""];
+        $user = auth()->user();
+        $video = Product::where('id', $request->product_id)->where('user_id', $user->id)->first();
+        if ($user->cannot('delete', $video)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+        $response = $this->productService->deleteVideo($request);
+
         return response()->json($response);
     }
 
-    public function deleteProductVideo(Request $req)
+    /**
+     * Update product sorting by logged brand
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function reorderProduct(Request $request): JsonResponse
     {
-        Video::where('id', $req->id)->delete();
-        $response = ['res' => true, 'msg' => "Deleted Successfully", 'data' => ""];
+        $user = auth()->user();
+        if ($user->cannot('viewAny', Product::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+        $response = $this->productService->reorderProduct($request);
 
         return response()->json($response);
     }
 
-    public function updateProductsStock(Request $request)
+    /**
+     * Inventory stock by product
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function updateStock(Request $request): JsonResponse
     {
-        if ($request->variant_id) {
-            ProductVariation::where('id', $request->variant_id)->update(array('stock' => $request->stock));
-        } else {
-            Products::where('id', $request->input('id'))->update(array('stock' => $request->stock));
+        $user = auth()->user();
+        if ($user->cannot('viewAny', Product::class)) {
```

Comment:
Why update is checked with 'viewAny' and validation is missing

Reply:
Done